src/util/feefrac.h

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_FEEFRAC_H
+#define BITCOIN_UTIL_FEEFRAC_H
+
+#include <stdint.h>
+#include <compare>
+#include <vector>
+#include <span.h>
+#include <util/check.h>
+
+/** Data structure storing a fee and size, ordered by increasing fee/size.
+ *
+ * The size of a FeeFrac cannot be zero unless the fee is also zero.
+ *
+ * FeeFracs have a total ordering, first by increasing feerate (ratio of fee over size), and then
+ * by decreasing size. The empty FeeFrac (fee and size both 0) sorts last. So for example, the
+ * following FeeFracs are in sorted order:
+ *
+ * - fee=0 size=1 (feerate 0)
+ * - fee=1 size=2 (feerate 0.5)
+ * - fee=2 size=3 (feerate 0.667...)
+ * - fee=2 size=2 (feerate 1)
+ * - fee=1 size=1 (feerate 1)
+ * - fee=3 size=2 (feerate 1.5)
+ * - fee=2 size=1 (feerate 2)
+ * - fee=0 size=0 (undefined feerate)
+ *
```

Comment:
In implementation they are sorted in increasing fee rate and believe the comments should be ordered like as is? and also the 0 fee and size chuck sorts first. 
```suggestion
 * by decreasing size. The empty FeeFrac (fee and size both 0) sorts first. So for example, the
 * following FeeFracs are in sorted order:
 *
 * - fee=0 size=0 (undefined feerate)
 * - fee=2 size=1 (feerate 2)
 * - fee=3 size=2 (feerate 1.5)
 * - fee=1 size=1 (feerate 1)
 * - fee=2 size=2 (feerate 1)
 * - fee=2 size=3 (feerate 0.667...)
 * - fee=1 size=2 (feerate 0.5)
 * - fee=0 size=1 (feerate 0)
```

Reply:
I don't understand your comment, sorry. They're listed in increasing feerate, with decreasing size as a tie breaker.

Reply:
Yes but in reverse I think.
Suppose to be the other way around no?

A sorted `FeeFracs`  should have undefined chunks first, the highest fee rate chunk second continuously...
If their is a tie, the chunk with lower size comes first.
Hence the are sorted in increasing fee rates, and then by decreasing size.

I've added test to verify this here https://github.com/ismaelsadeeq/bitcoin/commit/8ce89b132f4304a7feda067a88fd0a72330044a6, this test passed on this branch.

Reply:
Yes thanks for clarification 👍🏾

Reply:
I looked at it the other way around, using the result from `BuildDiagramFromUnsortedChunks` sort which custom > operator @instagibbs.